tests/unit_tests/conditional/cd/test_cd.py

```diff
@@ -63,48 +67,67 @@ def test_cd_tests_error(cd_estimator):
     with pytest.raises(RuntimeError, match="Group indications in"):
         cd_estimator.test(sample_df, {x}, {y}, group_col="group")
 
+    # test pre-fit propensity scores, or custom propensity model
+    with pytest.raises(
+        ValueError, match="Both propensity model and propensity estimates are specified"
+    ):
+        cd_estimator = cd_func(propensity_model=RandomForestClassifier(), propensity_est=[0.5, 0.5])
+        cd_estimator.test(sample_df, {x}, y_vars={y}, group_col="group")
+
+    with pytest.raises(ValueError, match="There are 3 group pre-defined estimates"):
+        cd_estimator = cd_func(propensity_est=np.ones((10, 3)) * 0.5)
+        cd_estimator.test(sample_df, {x}, y_vars={y}, group_col="group")
+
+    with pytest.raises(ValueError, match="There are 100 pre-defined estimates"):
+        cd_estimator = cd_func(propensity_est=np.ones((100, 2)) * 0.5)
+        cd_estimator.test(sample_df, {x}, y_vars={y}, group_col="group")
+
 
 @pytest.mark.parametrize(
     ["cd_func", "cd_kwargs"],
     [
         [BregmanCDTest, dict()],
         [KernelCDTest, dict()],
+        [BregmanCDTest, {"propensity_model": RandomForestClassifier()}],
+        [BregmanCDTest, {"propensity_est": np.ones((n_samples, 2)) * 0.5}],
+        [KernelCDTest, {"propensity_model": RandomForestClassifier()}],
+        [KernelCDTest, {"propensity_est": np.ones((n_samples, 2)) * 0.5}],
         [KernelCDTest, {"l2": 1e-3}],
         [KernelCDTest, {"l2": (1e-3, 2e-3)}],
     ],
 )
 @pytest.mark.parametrize(
     ["df", "env_type"],
     [
-        [single_env_scm(), "single"],
-        [multi_env_scm(), "multi"],
+        [single_env_scm(n_samples=n_samples, offset=2.0), "single"],
+        [multi_env_scm(n_samples=n_samples // 2, offset=2.0), "multi"],
     ],
 )
 def test_cd_simulation(cd_func, df, env_type, cd_kwargs):
     """Test conditional discrepancy tests."""
     random_state = 12345
-    print(cd_func, cd_kwargs)
-    cd_estimator = cd_func(random_state=random_state, null_reps=50, n_jobs=-1, **cd_kwargs)
+    cd_estimator = cd_func(random_state=random_state, null_reps=15, n_jobs=-1, **cd_kwargs)
 
     group_col = "group"
+    alpha = 0.1
 
     if env_type == "single":
         _, pvalue = cd_estimator.test(df, {"x"}, {"x1"}, group_col=group_col)
         print(pvalue)
-        assert pvalue > 0.05
+        assert pvalue > alpha
         _, pvalue = cd_estimator.test(df, {"x"}, {"z"}, group_col=group_col)
         print(pvalue)
```

Comment:
Should these prints be left in the test code?

Reply:
I left those in there to allow easier debugging of which one fails, but I am adding a better fix.